### PR TITLE
Active record 12to17

### PR DESCRIPTION
**Pull request by cowens87**

### CODE REVIEW

File: app/models/boba_tea.rb [hunk 1]
```diff
@@ -5,4 +5,16 @@ class BobaTea < ApplicationRecord
   
   validates_presence_of :name, :price, :base, :description
   validates_inclusion_of :caffeinated, in: [true, false]
+
+  def self.true_false
+    order(caffeinated: :desc)
+  end
+
+  def self.teas_by_price(amount)
+    where('price >= ?', amount)
```
Comment (eamouzou) [inline app/models/boba_tea.rb:14]: :nice_:

Comment (eamouzou) [inline app/models/boba_tea.rb:14]: nice

File: app/models/teahouse.rb [hunk 2]
```diff
@@ -1,4 +1,11 @@
 class Teahouse < ApplicationRecord
   validates_presence_of :name, :address, :city, :state, :zip
   has_many :boba_teas, dependent: :destroy
+
+   def self.sort_by_tea_count
```
Comment (eamouzou) [inline app/models/teahouse.rb:5]: curious: why the class method?

File: app/views/teahouse_boba_teas/index.html.erb [hunk 3]
```diff
@@ -1,13 +1,16 @@
 <h1>List of Boba Teas</h1>
 
-<%= link_to 'Add Boba Tea', new_teahouse_boba_teas_path %>
+<%= link_to 'Add Boba Tea', new_teahouse_boba_teas_path %><br>
+<%= link_to "Sort Tea Name by Alpha", "/teahouses/#{@teahouse.id}/boba_teas?sort=true", method: :get  %>
```
Comment (eamouzou) [inline app/views/teahouse_boba_teas/index.html.erb:4]: what's 'alpha'?

File: app/models/teahouse.rb (hunk 2 above)
Comment (cowens87) [reply app/models/teahouse.rb:5]: here is the user story associated with this method. I chose a class method b/c in my mind I thought all the teahouses tea counts should be looked at in order to sort.

Comment (cowens87) [reply app/models/teahouse.rb:5]: 16. [x] done >> User Story 16, Sort Teahouses by Number of Boba_tea

As a visitor
When I visit the Teahouses Index Page
Then I see a link to sort teahouses by the number of `boba_tea_table_name` they have
When I click on the link
I'm taken back to the Teahouse Index Page where I see all of the teahouses in order of their count of `boba_tea_table_name` (highest to lowest) And, I see the number of boba_tea next to each teahouse name

File: app/views/teahouse_boba_teas/index.html.erb (hunk 3 above)
Comment (cowens87) [reply app/views/teahouse_boba_teas/index.html.erb:4]: You're right, terrible naming, Updated to 'Sort by Tea Name'

**Pull request merged**